lib/matplotlib/tests/test_lines.py

```diff
@@ -199,6 +199,30 @@ def test_step_markers(fig_test, fig_ref):
     fig_ref.subplots().plot([0, 0, 1], [0, 1, 1], "-o", markevery=[0, 2])
 
 
+@check_figures_equal(extensions=('png',))
+def test_markevery(fig_test, fig_ref):
+    np.random.seed(42)
+    t = np.linspace(0, 3, 14)
+    y = np.random.rand(len(t))
+
+    casesA = [None, 4, (2, 5), [1, 5, 11],
```

Comment:
I think this would be more readable:

~~~
    cases = [
        (None, "11111111111111"),
        (4, "10001000100010"),
        ...
    ]

    for ax_test, ax_ref, (markevery, pattern) in zip(axs_test, axs_ref, cases):
        ax_test.plot(t, y, "-gD", markevery=markevery)
        ax_ref.plot(t, y, "-gD",
                    markevery=np.array(list(pattern)).astype(int).astype(bool))
~~~